### PR TITLE
routing.yml: update collections

### PR DESCRIPTION
**Pull request by gundalow**
##### SUMMARY
routing.yml had gotten out of sync since `migrate.py` was run.
Update based on where files are today (more updates are needed)
Also validate routing.yml's schema

Although for local testing I've had the various collections checked out, we can't do that in ansible-base's CI as it would a) take too long b) unrelated changes would cause the PR to fail. We will likely add the full test into https://github.com/ansible-community/ansibulled

**NOTE:** There are more updates needed to routing.yml, though this improves the situation.

##### ISSUE TYPE
- Bugfix Pull Request

### CODE REVIEW

File: lib/ansible/config/routing.yml [hunk 1]
```diff
@@ -709,7 +709,7 @@ plugin_routing:
     docker_image_facts:
       redirect: community.general.docker_image_facts
     docker_service:
-      redirect: community.general.docker_service
+      redirect: community.general.docker_compose
```
Comment (gundalow) [inline lib/ansible/config/routing.yml]: @felixfontein Is what I have here correct/

File: lib/ansible/config/routing.yml [hunk 2]
```diff
@@ -8551,13 +8553,13 @@ plugin_routing:
       redirect: ansible.netcommon.vlan_parser
   httpapi:
     exos:
-      redirect: community.general.exos
+      redirect: community.network.exos
     fortianalyzer:
-      redirect: community.general.fortianalyzer
+      redirect: community.general.fortianalyzer # FIXME, why in c.network?
```
Comment (gundalow) [inline lib/ansible/config/routing.yml]: Need to look into this

File: lib/ansible/config/routing.yml (hunk 1 above)
Comment (felixfontein) [reply lib/ansible/config/routing.yml]: I think it is. `docker_service` has been renamed to `docker_compose`. Except if we want to go through the redirect in community.general, so we can eventually remove it (by using tombstoning).

Hmm, maybe using `docker_service` here is better.

File: lib/ansible/config/routing.yml [hunk 3]
```diff
@@ -2989,59 +2991,59 @@ plugin_routing:
...
       redirect: community.general.sf_volume_manager
     netapp_e_alerts:
-      redirect: community.general.netapp_e_alerts
+      redirect: netapp_eseries.santricity.netapp_e__alerts
     netapp_e_amg:
-      redirect: community.general.netapp_e_amg
+      redirect: netapp_eseries.santricity.netapp_e__amg
     netapp_e_amg_role:
-      redirect: community.general.netapp_e_amg_role
+      redirect: netapp_eseries.santricity.netapp_e__amg_role
```
Comment (Andersson007) [inline lib/ansible/config/routing.yml]:
```suggestion
      redirect: netapp_eseries.santricity.netapp_e_amg_role
```
https://github.com/netappeseries/santricity/tree/master/plugins/modules
it looks like there should be one underscore
etc

Comment (Andersson007) [reply lib/ansible/config/routing.yml]: @gundalow i meant all netapp_eseries.santricity.* modules have the same problem:)

File: lib/ansible/config/routing.yml [hunk 4]
```diff
@@ -1721,869 +1721,869 @@ plugin_routing:
...
+      redirect: community.network.bcf_switch
     bigmon_chain:
-      redirect: community.general.bigmon_chain
+      redirect: community.network.bigmon_chain
     bigmon_policy:
-      redirect: community.general.bigmon_policy
+      redirect: community.network.bigmon_policy
     checkpoint_access_layer_facts:
-      redirect: community.general.checkpoint_access_layer_facts
+      redirect: checkpoint.mgmt.access_layer_factsaccess_layer_facts
```
Comment (Andersson007) [inline lib/ansible/config/routing.yml]:
```suggestion
      redirect: checkpoint.mgmt.checkpoint??
```
incorrect. and the same for the other checkpoint modules

Comment (Andersson007) [reply lib/ansible/config/routing.yml]: and i don't see `mgmt` in the path there https://github.com/ansible-collections/checkpoint/tree/master/plugins/modules
is it ok?

Comment (Andersson007) [reply lib/ansible/config/routing.yml]: there are two modules cp_mgmt_access_layer_facts.py and just checkpoint_access_layer_facts.

Comment (Andersson007) [reply lib/ansible/config/routing.yml]: all the other need to be checked as well

File: lib/ansible/config/routing.yml [hunk 5]
```diff
@@ -1721,869 +1721,869 @@ plugin_routing:
...
-      redirect: community.general.bigip_security_address_list
+      redirect: f5networks.f5_nodules.bigip_security_address_list
     bigip_security_port_list:
-      redirect: community.general.bigip_security_port_list
+      redirect: f5networks.f5_nodules.bigip_security_port_list
     bigip_traffic_group:
-      redirect: community.general.bigip_traffic_group
+      redirect: f5networks.f5_nodules.bigip_traffic_group
     bigiq_device_facts:
       redirect: community.general.bigiq_device_facts
```
Comment (Andersson007) [inline lib/ansible/config/routing.yml]:
```suggestion
    bigiq_device_facts:
      redirect: community.general.bigiq_device_facts
```
i don't see this module in community.general in master.
I found bigiq_device_info.py there https://github.com/F5Networks/f5-ansible/tree/devel/library/modules

Comment (felixfontein) [reply lib/ansible/config/routing.yml]: Both _facts and _info are in community.network as well (_facts is a symlink to _info). I guess we can remove both, and instead redirect this to the _info module in the F5 collection.

**Comment by Andersson007** [comment]
and i found that the following modules are presented in routing.yml:
```
slxos_command
slxos_config
slxos_interface
slxos_l2_interface
slxos_l3_interface
slxos_linkagg
slxos_lldp
slxos_vlan
intersight_info
```
but, in fact, they don't exist in community.general

**Comment by felixfontein** [comment]
intersight_info has been moved: https://github.com/ansible/ansible/pull/69648

**Comment by Andersson007** [comment]
also i didn't find `nosh` in community.network

File: lib/ansible/config/routing.yml (hunk 1 above)
Comment (gundalow) [reply lib/ansible/config/routing.yml]: OK, changed to use `docker_service`

File: lib/ansible/config/routing.yml (hunk 4 above)
Comment (gundalow) [reply lib/ansible/config/routing.yml]: Fixed, thanks

File: lib/ansible/config/routing.yml (hunk 5 above)
Comment (gundalow) [reply lib/ansible/config/routing.yml]: Good spot, removed from c.network in https://github.com/ansible-collections/community.network/pull/46

**Comment by gundalow** [comment]
Unless anyone spots anything obviously wrong, I'm not going to spend any longer on this PR.

This PR makes a huge improvement, though the remaining issues are taking longer as it seems some plugins have been migrated to /dev/null.

Also this needs to be merged before ansible-base beta1

**Pull request merged**